api/service.go

```diff
@@ -621,15 +777,22 @@ func (s *service) AddFile(server pb.API_AddFileServer) error {
 	pth, err := s.ipfsClient.Unixfs().Add(
 		server.Context(),
 		ipfsfiles.NewReaderFile(reader),
-		options.Unixfs.Pin(true),
+		options.Unixfs.Pin(false),
 		options.Unixfs.Progress(true),
 		options.Unixfs.Events(eventCh))
 	if err != nil {
 		return err
 	}
 
-	_, err = s.collections.Files.Create(server.Context(), pth, name, proj.ID)
-	if err != nil && !strings.Contains(err.Error(), store.ErrUniqueExists.Error()) {
+	dirpth, err := s.ipfsClient.Object().
```

Comment:
Add the link under the node. The trick here is to use to object API, not the Files API 🤷‍♂

Reply:
Huh?, that's weird. I wouldn't expect that adding a Link in some other API will have the same meaning in UnixFS. I mean, sounds reasonable but weird.

Sounds that would be natural to have some `UnixFS().Link(..)`. :/

Reply:
Agreed, yeah. Not very intuitive! `UnixFS` just has `Add`, `Get`, and `Ls`. `Add` doesn't let you manipulate existing nodes.